clients/src/main/java/org/apache/kafka/clients/producer/internals/TransactionManager.java

```diff
@@ -728,18 +779,11 @@ synchronized void maybeResolveSequences() {
                 } else {
                     // We would enter this branch if all in flight batches were ultimately expired in the producer.
                     if (isTransactional()) {
-                        // For the transactional producer, we bump the epoch if possible, otherwise we transition to a fatal error
+                        // For the transactional producer, we bump the epoch if possible, then transition to a fatal error
```

Comment:
It's a behaviour change here, when the transactional producer reaches this state, we'll do an epoch bump and then it'll be a fatal error.
Could you explain how it's changed actually? What's the difference between flipping the epochBumpRequired flag and go to abortable, and going to fatalbumpable?
Was the producer still usable after abortable transition (and the handled abort)?